Core/src/TrackFitting/GainMatrixUpdater.cpp

```diff
@@ -64,8 +64,9 @@ std::tuple<double, std::error_code> GainMatrixUpdater::visitMeasurement(
       return false;                                           // abort execution
     }
 
+    auto signCorrected = surface->correctSign(trackState.predicted);
```

Comment:
I think you should be able to do 
```cpp
trackState.referenceSurface()
```
to get the surface of the track state. This way, you don't need to pipe through the surface explicitly.

Reply:
I cannot find any surface method in `InternalTrackState` :thinking: 
```
/home/beomki/projects/acts/acts/Core/src/TrackFitting/GainMatrixUpdater.cpp:67:37: error: ‘struct Acts::GainMatrixUpdater::InternalTrackState’ has no member named ‘referenceSurface’
   67 |     auto signCorrected = trackState.referenceSurface()->correctSign(trackState.predicted);
```

Reply:
Yeah, that's an implementation detail of the gain matrix updater. Can you just add the surface pointer here, and fill it from the gain matrix updater template function with the pointer you get from the track state proxy?